crates/libs/core/src/runtime/package_change/config.rs

```diff
@@ -0,0 +1,168 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+//! Handle callbacks for configuration package changes
+//! TODO: We probably should also provide a helpful callback to use in conjunction with the config-rs support (so that it processes configuration changes)
+use mssf_com::FabricRuntime::{
+    IFabricCodePackageActivationContext6, IFabricConfigurationPackageChangeHandler,
+    IFabricConfigurationPackageChangeHandler_Impl,
+};
+
+use crate::runtime::config::ConfigurationPackage;
+
+use super::ConfigurationPackageChangeEvent;
+
+/// Rust trait to turn rust code into IFabricConfigurationPackageChangeHandler.
+/// Not exposed to user
+pub trait ConfigurationPackageChangeEventHandler: 'static {
+    fn on_change(&self, change: &ConfigurationPackageChangeEvent) -> crate::Result<()>;
+}
+
+// Bridge implementation for the change handler to turn rust code into SF com object.
+#[windows_core::implement(IFabricConfigurationPackageChangeHandler)]
+#[allow(non_camel_case_types)] // Suppress lint for _Impl struct
+pub struct ConfigurationPackageChangeEventHandlerBridge<T>
+where
+    T: ConfigurationPackageChangeEventHandler,
+{
+    inner: T,
+}
+
+impl<T> ConfigurationPackageChangeEventHandlerBridge<T>
+where
+    T: ConfigurationPackageChangeEventHandler,
+{
+    pub fn new(inner: T) -> Self {
+        Self { inner }
+    }
+}
+
+impl<T> IFabricConfigurationPackageChangeHandler_Impl
+    for ConfigurationPackageChangeEventHandlerBridge_Impl<T>
+where
+    T: ConfigurationPackageChangeEventHandler,
+{
+    fn OnPackageAdded(
+        &self,
+        _source: Option<&mssf_com::FabricRuntime::IFabricCodePackageActivationContext>,
+        configpackage: Option<&mssf_com::FabricRuntime::IFabricConfigurationPackage>,
+    ) {
+        let new_package = ConfigurationPackage::from_com(configpackage.unwrap().clone());
+        let event = ConfigurationPackageChangeEvent::Addition { new_package };
+        // TODO: unwrap, or should we change the return type of the lambda to be the empty type?
```

Comment:
Thoughts?

Reply:
Keep aligned with the COM api. SF does not handle failure return, so Rust api should match it. i.e. no return type

Reply:
Fixed